src/ShallowTraversal.js

```diff
@@ -110,7 +110,7 @@ export function buildPredicate(selector) {
           return node => nodeHasId(node, selector.slice(1));
 
         case SELECTOR.PROP_TYPE: {
-          const propKey = selector.split(/\[([a-zA-Z-]*?)(=|])/)[1];
+          const propKey = selector.split(/\[([a-zA-Z-0-9-]*?)(=|])/)[1];
```

Comment:
please also include tests that show that an attribute named `1-foo` would be invalid, per https://github.com/airbnb/enzyme/issues/693#issuecomment-261472688